jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt

```diff
@@ -98,8 +103,11 @@ class CawsConnectionProvider : GatewayConnectionProvider {
             return null
         }
 
+        val ssoSettings = connectionParams.ssoSettings ?: SsoSettings(SONO_URL, SONO_REGION)
+
         val connectionSettings = try {
-            CodeCatalystCredentialManager.getInstance().getSettingsAndPromptAuth()
+            loginSso(null, ssoSettings.startUrl, ssoSettings.region, CODECATALYST_SCOPES)
```

Comment:
does this do the right thing if you're currently logged into builder id, but the requested environment lies in an SSO tenant?

Reply:
Yes. loginSso handles it by default. It will sign you out of the BuilderID connection and sign you into Idc based on the parameters.